### PR TITLE
Fix membership auto-enroll webhooks

### LINKED ISSUE #2568: When the Topic is set to "Enrollment created" in WordPress Dashboard > LifterLMS > Settings > REST API > Webhooks, the payload only contains information on 1 course (even though there are more than one enrollments that occurred).
### Reproduction Steps
1. Make at LifterLMS is the only plugin installed.
2. Login to the WordPress site as an administrator.
3. Create a membership called `Membership 1`.
4. Create three courses called `Course 1`, `Course 2`, and `Course 3`.
5. Open `Membership 1` in the backend and add `Course 1`, `Course 2`, and `Course 3` to `Membership Settings > Auto Enrollment`.
6. Add a free access plan to `Membership 1`. 
7. Create a webhook in `WordPress Dashboard > LifterLMS > Settings > REST API > Webhooks` and set `Topic` to `Enrollment created`.
8. Use a free service like [webhook.site](https://webhook.site/) to generate the `Delivery URL` and to view the payload.
9. Create a student account in `WordPress Dashboard > Users > Add New`.
10. Log in as that student you created, and enroll on `Membership 1` while logged in as that student.
11. Got to [webhook.site](https://webhook.site/) to look at the payload.

### Expected Behavior
The payload should contain information about `Course 1`, `Course 2`, and `Course 3`.

### Actual Behavior
The payload contains information about `Course 1`.

This issue has be recreated:
+ [X] Locally
+ [] On a staging site
+ [X] On a production website
+ [X] With only LifterLMS and a default theme

### PR DESCRIPTION
**Pull request by brianhogg**
## Description

Currently there is a check to avoid the same webhook firing multiple times, but it's only checking the first argument or ID. For webhooks like the one that fires for the `enrollment.created` topic, there's two arguments: the student ID and the product ID (course or membership). Because of this, for an enrollment in a membership with one or more auto enrolled courses, only the webhook for the first course fires.

This PR changes the logic to look at the entire argument array when determining if there is a duplicate webhook. However from testing, it appears that the Action Scheduler will not allow a duplicate entry to be added anyway, so the check can likely be removed entirely.

Fixes https://github.com/gocodebox/lifterlms/issues/2568

## How has this been tested?

Automated tests, and manually [with the instructions on the ticket](https://github.com/gocodebox/lifterlms/issues/2568).

## Screenshots 

Captura de pantalla 2023-11-15 a las 4 34 07 p  m
Captura de pantalla 2023-11-15 a las 4 34 14 p  m
Captura de pantalla 2023-11-15 a las 4 34 21 p  m
Captura de pantalla 2023-11-15 a las 4 34 28 p  m

## Types of changes

Bug fix

## Checklist:
- [x] This PR requires and contains at least one changelog file. 
- [x] My code has been tested.
- [x] My code passes all existing automated tests. 
- [x] My code follows the LifterLMS Coding & Documentation Standards.

### CODE REVIEW

**Comment by kimcoleman** [comment]
@brianhogg thank you for this code we will run some tests.

Could you rebase this PR to dev branch?

**Comment by brianhogg** [comment]
> @brianhogg thank you for this code we will run some tests.
> 
> Could you rebase this PR to dev branch?

@kimcoleman done! :)

**Comment by actuallyakash** [comment]
hey @brianhogg 

Thanks for the code and the tests. Looks good. 👌 
Gonna close my PR in favour of this PR.

**Comment by brianhogg** [comment]
@actuallyakash I've removed the `is_already_processed` check and an associated test that was only checking if that check was happening in `should_deliver`.

It was a bit of luck when I made `is_already_processed` return `false` to verify the one test would pass, then noticed the new `test_multiple_hooks_for_single_webhook_only_schedules_once` test was also still passing :)

File: includes/models/class-llms-rest-webhook.php [hunk 1]
```diff
@@ -362,10 +340,6 @@ public function process_hook( ...$args ) {
 			return false;
 		}
 
-		// Mark this hook's first argument as processed to ensure it doesn't get processed again within the current request,
-		// as it might happen with webhooks with multiple hookes defined in `LLMS_REST_Webhooks::get_hooks()`.
-		$this->processed[] = $args;
```
Comment (actuallyakash) [inline includes/models/class-llms-rest-webhook.php]: Just need a docblock update for this.

File: includes/models/class-llms-rest-webhook.php [hunk 2]
```diff
@@ -434,8 +407,7 @@ protected function should_deliver( $args ) {
 
 		$deliver = ( 'active' === $this->get( 'status' ) ) // Must be active.
 			&& $this->is_valid_action( $args ) // Valid action.
-			&& $this->is_valid_resource( $args ) // Valid resource.
-			&& ! $this->is_already_processed( $args ); // Not already processed.
```
Comment (actuallyakash) [inline includes/models/class-llms-rest-webhook.php:435]: docblock update needed.

**Review by actuallyakash — CHANGES_REQUESTED** [review]

**Comment by actuallyakash** [comment]
Hey @brianhogg 

> @actuallyakash I've removed the `is_already_processed` check and an associated test that was only checking if that check was happening in `should_deliver`.

Cool, thanks. Just need to update some doc blocks.

> It was a bit of luck when I made `is_already_processed` return `false` to verify the one test would pass, then noticed the new `test_multiple_hooks_for_single_webhook_only_schedules_once` test was also still passing :)

oh, nice :D 

Just putting here for discussion, the `is_already_processed()` check was introduced due to this issue: https://github.com/gocodebox/lifterlms-rest/issues/210

However, the issue was due to the `pending_delivery` check which stopped the new webhooks from being delivered if the old ones were stuck. The `pending_delivery` check was removed and `is_already_processed` was introduced.

From the current PR, we now know that `is_already_processed` is not useful as "Action Scheduler does not seem to allow scheduling two actions with the same arguments and the same webhook ID".

File: includes/models/class-llms-rest-webhook.php (hunk 1 above)
Comment (brianhogg) [reply includes/models/class-llms-rest-webhook.php]: Done!

File: includes/models/class-llms-rest-webhook.php (hunk 2 above)
Comment (brianhogg) [reply includes/models/class-llms-rest-webhook.php:435]: Also done :)

**Comment by brianhogg** [comment]
> Cool, thanks. Just need to update some doc blocks.
> 

Docblocks updated, let me know if there's anything else or the updates aren't clear :)

> Just putting here for discussion, the `is_already_processed()` check was introduced due to this issue: #210
> 
> However, the issue was due to the `pending_delivery` check which stopped the new webhooks from being delivered if the old ones were stuck. The `pending_delivery` check was removed and `is_already_processed` was introduced.
> 
> From the current PR, we now know that `is_already_processed` is not useful as "Action Scheduler does not seem to allow scheduling two actions with the same arguments and the same webhook ID".

Thanks for the background! It doesn't look like this change will cause a regression. There's also now a test to ensure multiple actions can be scheduled for the same webhook in the same request, but with different arguments.

**Pull request merged**